planner/core/find_best_task.go

```diff
@@ -2440,14 +2650,30 @@ func (ds *DataSource) getOriginalPhysicalIndexScan(prop *property.PhysicalProper
 	return is
 }
 
-func (p *LogicalCTE) findBestTask(prop *property.PhysicalProperty, _ *PlanCounterTp, _ *physicalOptimizeOp) (t task, cntPlan int64, err error) {
+func (p *LogicalCTE) findBestTask(prop *property.PhysicalProperty, counter *PlanCounterTp, pop *physicalOptimizeOp) (t task, cntPlan int64, err error) {
+	if len(p.children) > 0 {
+		return p.baseLogicalPlan.findBestTask(prop, counter, pop)
```

Comment:
when should we set the children for LogicalCTE（child is field that it already has，we didn't utilize it before？）

Reply:
Currently, we use `whether it has children` to identify whether is producer or consumer.

Reply:
got it, better comment it above